targets/TARGET_STM/TARGET_STM32F3/device/stm32f3xx_hal_crc.c

```diff
@@ -454,6 +454,15 @@ HAL_CRC_StateTypeDef HAL_CRC_GetState(CRC_HandleTypeDef *hcrc)
   * @{
   */
 
+#if __GNUC__
```

Comment:
what is this solving for GCC?

Reply:
https://github.com/pauluap/mbed-os/commit/27425bca0a30db71b9366875fb46a02dde10d5c4

The warning messages that this change clears is shown in the commit message.  The warning only appears at the `-O3` optimization level